src/dotnet/CommandResolution/LockFileTargetExtensions.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Packaging.Core;
+using NuGet.ProjectModel;
+
+namespace Microsoft.DotNet.Cli.CommandResolution
+{
+    internal static class LockFileTargetExtensions
+    {
+        public static bool IsPortable(this LockFileTarget lockFileTarget)
+        {
+            return string.IsNullOrEmpty(lockFileTarget.RuntimeIdentifier) &&
+                lockFileTarget.GetPlatformLibrary() != null;
+        }
+
+        public static LockFileTargetLibrary GetPlatformLibrary(this LockFileTarget lockFileTarget)
+        {
+            // TODO: https://github.com/dotnet/sdk/issues/17 get this from the lock file
+            var platformPackageName = "Microsoft.NETCore.App";
+            var platformLibrary = lockFileTarget
+                .Libraries
+                .FirstOrDefault(e => e.Name.Equals(platformPackageName, StringComparison.OrdinalIgnoreCase));
+
+            return platformLibrary;
+        }
+
+        public static HashSet<string> GetPlatformExclusionList(
+            this LockFileTarget lockFileTarget,
+            IDictionary<string, LockFileTargetLibrary> libraryLookup)
+        {
+            var platformLibrary = lockFileTarget.GetPlatformLibrary();
+            var exclusionList = new HashSet<string>();
+
+            exclusionList.Add(platformLibrary.Name);
+            CollectDependencies(libraryLookup, platformLibrary.Dependencies, exclusionList);
+
+            return exclusionList;
+        }
+
+        public static IEnumerable<LockFileTargetLibrary> GetRuntimeLibraries(this LockFileTarget lockFileTarget)
+        {
+            IEnumerable<LockFileTargetLibrary> runtimeLibraries = lockFileTarget.Libraries;
+            Dictionary<string, LockFileTargetLibrary> libraryLookup =
+                runtimeLibraries.ToDictionary(e => e.Name, StringComparer.OrdinalIgnoreCase);
+
+            HashSet<string> allExclusionList = new HashSet<string>();
+
+            if (lockFileTarget.IsPortable())
+            {
+                allExclusionList.UnionWith(lockFileTarget.GetPlatformExclusionList(libraryLookup));
+            }
+
+            // TODO: exclude "type: build" dependencies during publish - https://github.com/dotnet/sdk/issues/42
```

Comment:
Probably remove these, if we aren't going to support "type: build" dependencies for CommandResolution.